### ISSUE TITLE
Crash on 10.9 with customized preferences

### ISSUE THREAD

**Issue by jbaker10** [ygini/Hello-IT#19]
Hey Yoann,

We are seeing on a machine running 10.95 (and confirmed on a VM running 10.9.5) that when using our customized preferences (under `/Library/Preferences/com.github.ygini.Hello-IT.plist`), the app crashes consistently. If you remove this file, or uninstall and reinstall Hello-IT fresh from Github, it will work fine.

The crash logs are as follows:

```
Jul 21 13:44:49 admin-Mac.local Hello IT[264]: Reachability Flag Status: R -----l- networkStatusForFlags
Jul 21 13:44:49 admin-Mac.local WindowServer[103]: _CGXGLDisplayContextForDisplayDevice: No matching context for device (0x7fbca1c12220) - disabling OpenGL
Jul 21 13:44:49 --- last message repeated 20 times ---
Jul 21 13:44:49 admin-Mac.local Hello IT[264]: -[__NSCFString containsString:]: unrecognized selector sent to instance 0x7f9cc2059200
Jul 21 13:44:49 admin-Mac.local Hello IT[264]: An uncaught exception was raised
Jul 21 13:44:49 admin-Mac.local Hello IT[264]: -[__NSCFString containsString:]: unrecognized selector sent to instance 0x7f9cc2059200
Jul 21 13:44:49 admin-Mac.local Hello IT[264]: (
        0   CoreFoundation                      0x00007fff8edb925c __exceptionPreprocess + 172
        1   libobjc.A.dylib                     0x00007fff8d70be75 objc_exception_throw + 43
        2   CoreFoundation                      0x00007fff8edbc12d -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
        3   CoreFoundation                      0x00007fff8ed17272 ___forwarding___ + 1010
        4   CoreFoundation                      0x00007fff8ed16df8 _CF_forwarding_prep_0 + 120
        5   TestHTTP                            0x000000010a760b17 TestHTTP + 6935
        6   Foundation                          0x00007fff8cd40d3d __67+[NSURLConnection sendAsynchronousRequest:queue:completionHandler:]_block_invoke_2 + 130
        7   Foundation                          0x00007fff8cbca745 -[NSBlockOperation main] + 75
        8   Foundation                          0x00007fff8cba9f31 -[__NSOperationInternal _start:] + 631
        9   Foundation                          0x00007fff8cba9bdb __NSOQSchedule_f + 64
        10  libdispatch.dylib                   0x00007fff946f428d _dispatch_client_callout + 8
        11  libdispatch.dylib                   0x00007fff946fbef0 _dispatch_main_queue_callback_4CF + 333
        12  CoreFoundation                      0x00007fff8ed204f9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
        13  CoreFoundation                      0x00007fff8ecdb714 __CFRunLoopRun + 1636
        14  CoreFoundation                      0x00007fff8ecdae75 CFRunLoopRunSpecific + 309
        15  HIToolbox                           0x00007fff95e6fa0d RunCurrentEventLoopInMode + 226
        16  HIToolbox                           0x00007fff95e6f7b7 ReceiveNextEventCommon + 479
        17  HIToolbox                           0x00007fff95e6f5bc _BlockUntilNextEventMatchingListInModeWithFilter + 65
        18  AppKit                              0x00007fff9362324e _DPSNextEvent + 1434
        19  AppKit                              0x00007fff9362289b -[NSApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 122
        20  AppKit                              0x00007fff9361699c -[NSApplication run] + 553
        21  AppKit                              0x00007fff93601783 NSApplicationMain + 940
        22  libdyld.dylib                       0x00007fff902705fd start + 1
    )
Jul 21 13:44:49 admin-Mac.local Hello IT[264]: *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString containsString:]: unrecognized selector sent to instance 0x7f9cc2059200'
    *** First throw call stack:
    (
        0   CoreFoundation                      0x00007fff8edb925c __exceptionPreprocess + 172
        1   libobjc.A.dylib                     0x00007fff8d70be75 objc_exception_throw + 43
        2   CoreFoundation                      0x00007fff8edbc12d -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
        3   CoreFoundation                      0x00007fff8ed17272 ___forwarding___ + 1010
        4   CoreFoundation                      0x00007fff8ed16df8 _CF_forwarding_prep_0 + 120
        5   TestHTTP                            0x000000010a760b17 TestHTTP + 6935
        6   Foundation                          0x00007fff8cd40d3d __67+[NSURLConnection sendAsynchronousRequest:queue:completionHandler:]_block_invoke_2 + 130
        7   Foundation                          0x00007fff8cbca745 -[NSBlockOperation main] + 75
        8   Foundation                          0x00007fff8cba9f31 -[__NSOperationInternal _start:] + 631
        9   Foundation                          0x00007fff8cba9bdb __NSOQSchedule_f + 64
        10  libdispatch.dylib                   0x00007fff946f428d _dispatch_client_callout + 8
        11  libdispatch.dylib                   0x00007fff946fbef0 _dispatch_main_queue_callback_4CF + 333
        12  CoreFoundation                      0x00007fff8ed204f9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
        13  CoreFoundation                      0x00007fff8ecdb714 __CFRunLoopRun + 1636
        14  CoreFoundation                      0x00007fff8ecdae75 CFRunLoopRunSpecific + 309
        15  HIToolbox                           0x00007fff95e6fa0d RunCurrentEventLoopInMode + 226
        16  HIToolbox                           0x00007fff95e6f7b7 ReceiveNextEventCommon + 479
        17  HIToolbox                           0x00007fff95e6f5bc _BlockUntilNextEventMatchingListInModeWithFilter + 65
        18  AppKit                              0x00007fff9362324e _DPSNextEvent + 1434
        19  AppKit                              0x00007fff9362289b -[NSApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 122
        20  AppKit                              0x00007fff9361699c -[NSApplication run] + 553
        21  AppKit                              0x00007fff93601783 NSApplicationMain + 940
        22  libdyld.dylib                       0x00007fff902705fd start + 1
    )
Jul 21 13:44:49 admin-Mac.local WindowServer[103]: _CGXGLDisplayContextForDisplayDevice: No matching context for device (0x7fbca1c12220) - disabling OpenGL
Jul 21 13:44:49 --- last message repeated 14 times ---
Jul 21 13:44:49 admin-Mac com.apple.launchd.peruser.501[166] (com.github.ygini.Hello-IT.28192[264]): Job appears to have crashed: Abort trap: 6
Jul 21 13:44:49 admin-Mac.local ReportCrash[246]: Saved crash report for Hello IT[264] version 1.2.1 (94) to /Users/admin/Library/Logs/DiagnosticReports/Hello IT_2016-07-21-admin-Mac.crash
Jul 21 13:44:49 admin-Mac.local WindowServer[103]: _CGXGLDisplayContextForDisplayDevice: No matching context for device (0x7fbca1c12220) - disabling OpenGL
Jul 21 13:44:55 --- last message repeated 22 times ---
Jul 21 13:44:55 admin-Mac.local WindowServer[103]: _CGXGLDisplayContextForDisplayDevice: No matching context for device (0x7fbca1d29310) - disabling OpenGL
```

I have tried removing each different piece within the plist to try and see if it was a specific plugin/site/application causing it, but it didn't make a difference.

Any thoughts?

**Comment by ygini** [comment]
Thanks for the crash report!

I've fixed this bug, update to 1.2.3 and you should be OK

https://github.com/ygini/Hello-IT/releases/tag/v1.2.3

**Comment by jbaker10** [comment]
Thanks @ygini for being so quick on this! Really appreciate the help and product!
